pallets/asset-manager/src/lib.rs

```diff
@@ -256,18 +261,26 @@ pub mod pallet {
 			metadata: <T::AssetConfig as AssetConfig>::AssetRegistrarMetadata,
 		) -> DispatchResult {
 			T::ModifierOrigin::ensure_origin(origin)?;
+			ensure!(
+				!asset_id.is_one(),
```

Comment:
The logic only works for a specific configuration.
Should check `AssetConfig<Config>::NativeAssetId` instead of `one`.